hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/Consumer.java

```diff
@@ -36,6 +37,7 @@
     private Subscription subscription;
 
     private volatile boolean consuming = true;
+    private volatile CountDownLatch stoppedLatch = new CountDownLatch(1);
```

Comment:
Volatile keyword is not needed here, since you never reassign stoppedLatch reference and CountDownLatch.countDown() operation is visible to other threads, as it *happens-before* CountDownLatch.await(). You may make this field final by the way.

Source: http://docs.oracle.com/javase/8/docs/api/java/util/concurrent/package-summary.html#MemoryVisibility